MatrixSDK/LocationSharing/MXBeaconInfoSummary.swift

```diff
@@ -0,0 +1,96 @@
+// 
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// MXBeaconInfoSummary summarize live location sharing for a user sharing his location in a room.
+@objcMembers
+public class MXBeaconInfoSummary: NSObject, MXBeaconInfoSummaryProtocol {
+    
+    // MARK: - Properties
+    
+    public let identifier: String
+    public let userId: String
+    public private(set) var deviceId: String?
+    public private(set) var beaconInfo: MXBeaconInfo
+    public private(set) var lastBeacon: MXBeacon?
+        
+    public var expiringDate: UInt64 {
+        return beaconInfo.timestamp + beaconInfo.timeout
+    }
+    
+    public var hasExpired: Bool {
+        return Date().timeIntervalSince1970 > TimeInterval(expiringDate)
```

Comment:
Maybe irrelevant based on my first comment about `expiringDate`, but just in case: `TimeInterval` expects its value in seconds, but I think `expiringDate` is represented in milliseconds? I wonder if it would be worth adding a couple of tests for expiry anyway to be doubly sure.